pkg/apis/numaflow/v1alpha1/pipeline_types.go

```diff
@@ -601,6 +601,8 @@ type PipelineStatus struct {
 	SourceCount *uint32       `json:"sourceCount,omitempty" protobuf:"varint,6,opt,name=sourceCount"`
 	SinkCount   *uint32       `json:"sinkCount,omitempty" protobuf:"varint,7,opt,name=sinkCount"`
 	UDFCount    *uint32       `json:"udfCount,omitempty" protobuf:"varint,8,opt,name=udfCount"`
+	// ObservedGeneration stores the generation value observed when setting the current Phase. Default value is -1
```

Comment:
`* observed by the controller.`

It should not go with the `phase`. When there's a generation change, e.g. a `patch` operation, the `generation` increases 1, but the `phase` goes with `pausing` and `paused`.

Do not need to mention default, using defaults 0 is good enough.